_includes/langselect.html

```diff
@@ -1,17 +1,39 @@
-<div id="langselect">
-  <div lang="en" id="suglang">This site might also be available in your native language. Select your preferred language below:</div>
-  <ul>
-    {% for lng in site.languages %}
-      {% if lng == site.active_lang %}
-        <li class="selected" data-slang="{{ lng }}">{{ lng }}</li>
-      {% else %}
-        {% if lng == site.default_lang %}
-          {% capture langlink %} {{ page.url }}{% endcapture %}
-        {% else %}
-          {% capture langlink %}/{{ lng }}{{ page.url }}{% endcapture %}
-        {% endif %}
-        <li><a href="{{ langlink }}" data-slang="{{ lng }}">{{ lng }}</a></li>
-      {% endif %}
-    {% endfor %}
-  </ul>
+<div id="langselect-container">
+  <div id="langselect">
+    <!-- Language Dropdown inside a form -->
+    <form action="" method="get" id="language-form">
+      <div class="language-dropdown">
+        <select id="language-select" name="language-select" onchange="location = this.value;">
+          {% assign sorted_languages = site.languages | sort %}
+          {% for lng in sorted_languages %}
+            {% if lng == site.default_lang %}
+              {% capture langlink %} {{ page.url }}{% endcapture %}
+              {% capture lang_rel %}canonical{% endcapture %}
+            {% else %}
+              {% capture langlink %}{{ site.baseurl }}/{{ lng }}{{ page.url }}{% endcapture %}
+              {% capture lang_rel %}alternate{% endcapture %}
+            {% endif %}
+            <option value="{{ langlink }}" {% if lng == site.active_lang %}selected{% endif %}>
```

Comment:
@gilgongo I'm not sure if/how something has to be added here for the canonical links.

Reply:
As long as the markup shows any link that goes to a `site.active_lang` page as `canonical` and (I think less importantly) any other langs as `alternate`, we're OK.

Reply:
Whoops, @ignotus666 sorry I meant `site.default_lang` (if that's a thing?)

Reply:
Only just got round to looking properly.

The markup seems to be coming out as `lang="canonical"` and `lang="alternate"` here when I view it with Jekyll. They need to be `hreflang="[lang]"` with the actual lang code of the URL. 

When the currently selected language isn't `site.default_lang`, URLs need to have `rel="alternate"`, otherwise they are `rel="canonical"` (which I think is happening?)

Reply:
@gilgongo did you test what happens with JS disabled? Does it work as intended then?

Reply:
Most of it seems OK with and without JS (eg at the top of the page) but some links like this in the nav (I think - hard to tell) are showing`<ul><li><a href="/wiki/Getting-Started" lang="canonical" class="">Setting Up</a><ul>` when they should be `rel="canonical"` and have `hreflang` tags.

Reply:
Could you test [this](https://github.com/ignotus666/jamuluswebsite/actions/runs/9875448341) version please? Sorry, I don't know what I should be looking for...

Reply:
That version (I've managed to un-minify it now) is almost OK but the nav seems to have slipped up a bit. It should be ` hreflang="es" ` here (not ` lang="alternate"`) with `rel="alternate"` added. 

```
<h3>Cómo usar Jamulus</h3>
  <ul>
   <li>
   <a href="/es/wiki/Getting-Started" lang="alternate" class="">Configuración</a>
   <ul>
   <li><a href="/es/wiki/Installation-for-Windows" lang="alternate" class="">Windows</a></li>
   <li><a href="/es/wiki/Installation-for-Macintosh" lang="alternate" class="">macOS</a></li>

```

Otherwise I think all the other links (eg the ones in the `<head>` section are OK. Essentially, en needs to **always** have `rel="canonical"`, since that (I hope!) will fix the issue of Google being confused when it finds the same URL given as a different language that has not (or has only partially) been translated. After that, all other languages need to identify their links with `rel="alternate"`.

BTW seems to be a glitch in the selector's `<noscript>` section maybe? For example ES is identifying as EN here:

```
<noscript>
    <ul class="language-list">
    <li><a href="/de/" rel="alternate" hreflang="de" class=""> de </a></li>
    <li><a href="/es/" rel="canonical" hreflang="en" class=""> en </a></li>
    <li><a href="/es/" rel="alternate" hreflang="es" class="active"> es </a></li>
    <li><a href="/fr/" rel="alternate" hreflang="fr" class=""> fr </a></li>
```

Reply:
@gilgongo according to [this](https://developers.google.com/search/docs/crawling-indexing/consolidate-duplicate-urls), the canonical link reference should be in the <head> section, which in our case is in mainhomepage.html. I tried adding the canonical stuff to langselect.html with an include in mainhomepage.html to avoid cluttering it up but that resulted in a duplicate dropdown button at the top of the page. So I created a separate lang-links.html file and that seems to work.

Could you please test [this](https://github.com/ignotus666/jamuluswebsite/actions/runs/9935684488) version?

Reply:
> BTW seems to be a glitch in the selector's <noscript> section maybe? For example ES is identifying as EN here:

That happens with whichever the active language is - my guess is that it's supposed to happen.

Reply:
> That happens with whichever the active language is - my guess is that it's supposed to be that way.

I don't think so. I'd consider this to be a bug.

Reply:
Also, in the current version you cannot click the language you've currently selected - this makes sense for the non JS version.

Reply:
@ann0see Ok, it's simple enough for non-JS mode, but seems to be tricky for JS mode without a script. Is JS mode good as it is or do we also want the currently selected language to be non-clickable in the dropdown?

Reply:
Ok done. Artifact is at the bottom.

Reply:
> I don't think so. I'd consider this to be a bug.

@ann0see I can't for the life of me figure out how to stop it doing that. Nothing I try changes that behaviour.

Reply:
I think this is fixed now.